core/src/main/java/org/infinispan/transaction/impl/LocalTransaction.java

```diff
@@ -103,9 +101,7 @@ public boolean isImplicitTransaction() {
 
    @Override
    public void putLookedUpEntry(Object key, CacheEntry e) {
-      if (isMarkedForRollback()) {
-         throw new CacheException("This transaction is marked for rollback and cannot acquire locks!");
-      }
+      checkIfRolledBack();
```

Comment:
I'm not sure this check is needed at all. I'd rather check only at the points where there are external effects, i.e. locks are acquired or entries are committed.

Reply:
`isMarkedForRollback()` is true when a `RollbackCommand` is executed. I think it is fine to be here.

Reply:
I wasn't saying it breaks anything, I was just thinking that it complicates the code paths for no good reason: you still have to check that the transaction was not rolled back before committing.

Not a priority, just me thinking aloud.